### ISSUE TITLE
problem when training on ml-1m

### ISSUE THREAD

**Issue by ShaoPengyang** [joeybose/Flexible-Fairness-Constraints#1]
we are trying to operate an experiment on your codes.
Dataset--movieslens-1M, however, when training modelD and filters(these two trained together),we get a loss which value below 0.
is that normal?

**Comment by ShaoPengyang** [comment]
here's my setting. (1)dataset:ml-1m, (2)model:simple-gcmc, (3)--use_cross_entropy(4)batch_size 8192(5)gamma 10 (6)embed_dim 30, lr 0.01 --p 1.
And, we found that in 'transD_moiveslens.py',   the simpleGCMC's output is above 0, but fair_penalty is below 0, because of this code:
           ' if not args.use_cross_entropy:
                fair_penalty = constant - l_penalty
            else:
                fair_penalty = -1*l_penalty'
gamma = 10 ,so we get a full_loss under 0:
"
-> full_loss.backward(retain_graph=False)
(Pdb) full_loss
tensor(-35.9037, device='cuda:0', grad_fn=<AddBackward0>)
(Pdb) task_loss
tensor(14.6623, device='cuda:0', grad_fn=<NllLossBackward>)
(Pdb) fair_penalty
tensor(-5.0566, device='cuda:0', grad_fn=<MulBackward0>)
"

**Comment by ShaoPengyang** [comment]
![image](https://user-images.githubusercontent.com/37691345/71975628-65350480-324f-11ea-8bb3-572261f4aa8b.png)

**Comment by ShaoPengyang** [comment]
Also, in main_moivelens.py line 97,98
   ' args.train_ratings['movie_id'] += int(np.max(args.users['user_id']))
    args.test_ratings['movie_id'] += int(np.max(args.users['user_id']))'
leading to a result: movie_id 6039-9990(movie_id+0-6039)  user_id  0-6039
should be changed into 
   '
    args.train_ratings['movie_id'] += int(np.max(args.users['user_id']))+1
    args.test_ratings['movie_id'] += int(np.max(args.users['user_id']))+1
   '
you can check

**Comment by joeybose** [comment]
Hi, the indexing issue is a known bug that I forgot to upload the fixed version for. It shouldn't compromise the results. With regards to your loss observation, that is fairly standard in adversarial losses its always up and down and the gamma factor controls the strength of the adversarial objective so this is not too surprising.

**Comment by ShaoPengyang** [comment]
thank you.
